docs/configuration/listeners/lds.rst

```diff
@@ -0,0 +1,77 @@
+.. _config_listeners_lds:
+
+Listener discovery service
+==========================
+
+The listener discovery service (LDS) is an optional API that Envoy will call to dynamically fetch
+listeners. Envoy will reconcile the API response and add, modify, or remove known listeners
+depending on what is required.
+
+The semantics of listener updates are as follows:
+
+* Every listener must have a unique :ref:`name <config_listeners_name>`. If a name is not
+  provided, Envoy will create a UUID. Listeners that are to be dynamically updated should have a
+  unique name supplied by the management server.
+* When a listener is added, it will be "warmed" before taking traffic. For example, if the listener
+  references an :ref:`RDS <config_http_conn_man_rds>` configuration, that configuration will be
+  resolved and fetched before the listener is moved to "active."
```

Comment:
On Listener update, we make a whole new listener, which does a whole new RDS fetch. This will wait for the first response/failure before moving from warming to active. (I will clarify this also).

This part is really an implementation detail. For example, if a listeners RDS config references ADS, ADS could potentially cache. Better. @junr03 is going to add the HttpRouteManager t house RDS responses which could cache and decouple from the listeners.